lms/djangoapps/instructor/views/api.py

```diff
@@ -2821,21 +2821,38 @@ def start_certificate_regeneration(request, course_id):
     """
     course_key = CourseKey.from_string(course_id)
     certificates_statuses = request.POST.getlist('certificate_statuses', [])
+
     if not certificates_statuses:
         return JsonResponse(
             {'message': _('Please select one or more certificate statuses that require certificate regeneration.')},
             status=400
         )
 
     # Check if the selected statuses are allowed
-    allowed_statuses = [CertificateStatuses.downloadable, CertificateStatuses.error, CertificateStatuses.notpassing]
+    allowed_statuses = [
+        CertificateStatuses.downloadable,
+        CertificateStatuses.error,
+        CertificateStatuses.notpassing,
+        CertificateStatuses.audit_passing,
+        CertificateStatuses.audit_notpassing,
+    ]
     if not set(certificates_statuses).issubset(allowed_statuses):
         return JsonResponse(
             {'message': _('Please select certificate statuses from the list only.')},
             status=400
         )
+
+    student_set = None
+    if {CertificateStatuses.audit_passing, CertificateStatuses.audit_notpassing} & set(certificates_statuses):
+        student_set = 'verified_users_with_audit_certs'
+
     try:
-        instructor_task.api.regenerate_certificates(request, course_key, certificates_statuses)
+        instructor_task.api.regenerate_certificates(
+            request,
+            course_key,
+            certificates_statuses,
+            student_set=student_set
```

Comment:
let's leave out this `student_set` kwarg and just add `audit_passing` and `audit_notpassing` to `allowed_statuses` and call it a day. It's a hack, and we'll be doing a lot of unnecessary cert generation, but it will address this issue. Chalk this one up to tech debt in the certs project.